### ISSUE TITLE
Visualize Key IDs and compute fingerpritns

### ISSUE THREAD

**Issue by flowlo** [keybase/keybase-issues#1893]
Key IDs in their conservative form, i.e.

```
A2D6 BA02 EBB1 C984
```

are not easily comprehensible by the human mind. It is hard to draw context from a bunch of hexadecimal characters, which makes it hard to 
 * memorize them
 * compare them with high accuracy.

Secondly, I consider key IDs provided without fingerprints deprecated as filed in #1486 

Given that future versions of Keybase will likely show fingerprints instead of only key IDs, the problem of comprehension grows bigger:

```
03FA 933F 8D72 D752 AF8A  7D8F A2D6 BA02 EBB1 C984
```

I therefore propose to offer alternate means of rendering those hex numbers on Keybase profile pages.

These include, but are not limited to:

 1. QR code (as used in OpenKeychain)
 2. Colored Fingerprint in text form (color can be directly derived from hex representation of fingerprint)
 3. Rendered, colored and structured image derived from fingerprint.

As (1.) and (2.) are trivial, I'd like to give an example for (3.) taken from mailpile/Mailpile#1093:

![](https://cloud.githubusercontent.com/assets/3943592/6173221/34fead8e-b2de-11e4-962b-9a4cbe766af9.png)

These colored patterns are derived from the fingerprint. As (parts of) a V4 fingerprint according to [RFC 4880, section 12.2](https://tools.ietf.org/html/rfc4880#section-12.2) are hashed from the public key, using a color scheme derived from a slightly different (maybe having the same key ID) key would result in a completely different image (pattern and coloring). Looking at two of those images to compare them is significantly easier for humans, therefore would be a great increase in usability for Keybase.
Of course, the generation of those "rendered fingerprints" is to be discussed and should fit with the design language of Keybase as such.

**Comment by mitar** [comment]
+1

**Comment by mitar** [comment]
Not sure if those keyholes are the best though. The best would be if also the basic pattern/structure would change. So that it is not mostly just color (what about color blind people) and slight length variations. Humans observe patterns the most. Color and length comparisons not so much.

**Comment by flowlo** [comment]
@mitar Good point! I would like to bring up [Identicons](https://en.wikipedia.org/wiki/Identicon) as [used by GitHub](https://github.com/blog/1586-identicons) for example. A variation of that approach could be very well suited to visualize fingerprints. WDYT?
